src/cache_handler.js

```diff
@@ -3,10 +3,12 @@ const debug = require('debug')('bte:biothings-explorer-trapi:cache_handler');
 const LogEntry = require('./log_entry');
 
 module.exports = class {
-  constructor(qEdges, logs = []) {
+  constructor(qEdges, caching, logs = []) {
     this.qEdges = qEdges;
     this.logs = logs;
-    this.cacheEnabled = !(process.env.REDIS_HOST === undefined) && !(process.env.REDIS_PORT === undefined);
+    this.cacheEnabled = (caching !== undefined) ?
+      caching == 'true' : //convert string to bool
+      (!(process.env.REDIS_HOST === undefined) && !(process.env.REDIS_PORT === undefined));
```

Comment:
if `caching==true` but REDIS env variables are not defined, `this.cacheEnabled` should still be false, right?

Reply:
Right now, it just sets it to whatever the user puts, so making query like `POST /v1/query?caching=true` would set it to true regardless of the variables but I guess it wouldn't really make sense to set it to enable it if it won't work... The only thing is that it would be a little odd for the user to explicitly specify `caching=true` and the server executes it without caching.

Reply:
I think the practical use case is for users to have a way to disable caching for debugging (or sometimes they want to refresh results within 30min default caching lifetime), so we would expect users to pass either `caching=false` or not passing anything (which is the same as `caching=true`).

Reply:
In the case that it isn't passed, it would be `undefined` so it would check check the redis variables. So for the current code if the redis variables weren't defined, `POST /v1/query?caching=true` would be `true` but `POST /v1/query` would be `false`.

Reply:
Does this cover all scenarios and match with what you have in mind?

```
query.caching         true    false   undefined   true    false   undefined
REDIS env variables   yes       yes      yes       no       no        no

this.cacheEnabled     yes       no       yes       no       no        no
```

Reply:
Just to clarify, the current code does this:
```
query.caching         true    false   undefined   true    false   undefined
REDIS env variables   yes       yes      yes       no       no        no
this.cacheEnabled     yes       no       yes       yes       no        no
```

What I meant is that in the case of `query.caching=true` and `REDIS env variables=no` it might be a bit illogical for the user to specify caching to be on yet the cache would still be off, but now that I think of it I think it would be a little more logical to make it match your table so I will make those changes.

Reply:
yeah, this is like when the caching is not supported at the backend, then user's input will be ignored any way. Also in the response, there will be a log entry indicating whether the caching is actually used or not.